source/common/http/filter_manager.cc

```diff
@@ -371,6 +371,32 @@ void ActiveStreamDecoderFilter::onDecoderFilterAboveWriteBufferHighWatermark() {
   parent_.filter_manager_callbacks_.onDecoderFilterAboveWriteBufferHighWatermark();
 }
 
+FilterHeadersStatus ActiveStreamDecoderFilter::decodeHeaders(RequestHeaderMap& headers,
+                                                             bool end_stream) {
+  // Each decoder filter instance checks if the request passed to the filter is gRPC
+  // so that we can issue gRPC local responses to gRPC requests. Filter's decodeHeaders()
+  // called here may change the content type, so we must check it before the call.
+
+  is_grpc_request_ = Grpc::Common::isGrpcRequestHeaders(headers);
+  FilterHeadersStatus status = handle_->decodeHeaders(headers, end_stream);
+
+  // Validate filters did not erroneously remove required headers. If they do, send a direct
+  // response.
+  const Http::Status header_status = HeaderUtility::checkRequiredHeaders(headers);
```

Comment:
do we still need this here?  can't we just fail with this error if status above is not OK?

Reply:
Yep -- here's the scenario

Filter A removes path header
Filter B access path header to add to metadata -> crash.

So we verify requirements after making a call to untrusted extensions.